Source/API/creature.h

```diff
@@ -18,8 +18,8 @@ class creature {
 
     // TODO: Improve the ease of getting chromosomes
     template <class TChromosome> //TODO: Why does this have to be templated
-    inline TChromosome* get_chromosome(const std::string& name) {
-        chromosome* chromosome =  _chromosomes[name];
+    inline TChromosome* get_chromosome(const std::string& name) const {
+        chromosome* chromosome = _chromosomes.find(name)->second;
```

Comment:
Why did you switch this to use `_chromosomes.find(name)->second` instead of sticking with `_chromosomes[name]`?

Reply:
The access operator [] is not const and can modify the contents of the map. Therefore it can't be used in a const function. The find method is const and achieves the same behavior so it is a suitable replacement.